src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegrationInDirectory.tsx

```diff
@@ -140,11 +142,33 @@ export default class InstalledIntegrationInDirectory extends React.Component<Pro
               <IntegrationItem integration={integration} />
             </IntegrationItemBox>
             <div>
-              {
+              {showSlackAlert && (
                 <Tooltip
-                  disabled={this.hasConfiguration()}
-                  position="left"
-                  title="Integration not configurable"
+                  disabled={hasAccess}
+                  title={t(
+                    'You must be an organization owner, manager or admin to re-authenticate'
+                  )}
+                >
+                  <Button
+                    disabled={!hasAccess}
+                    priority="primary"
+                    size="small"
+                    icon={<IconWarning size="sm" />}
+                  >
+                    {t('Re-authenticate Now')}
+                  </Button>
+                </Tooltip>
+              )}
+              <Tooltip
+                disabled={this.hasConfiguration()}
+                position="left"
+                title="Integration not configurable"
+              >
+                <Tooltip
```

Comment:
Tooltips within toooltips?

Reply:
@EvanPurkhiser I guess I could just make the text configurable instead, what do you think?

Reply:
Well depends what happens if you have both errors

Reply:
One would take priority over the other